test/controllers/account/permissions_controller_test.rb

```diff
@@ -5,7 +5,7 @@ class Account::PermissionsControllerTest < ActionController::TestCase
     should "prevent unauthenticated users" do
       application = create(:application)
 
-      get :show, params: { application_id: application.id }
+      get :show, params: { application_id: application }
```

Comment:
Interesting! I thought this was a pretty common thing to do (in requests and/or method calls), and it's been around in Rails since at least 2020 (also struggling to find any proper documentation beyond the code itself):
- https://github.com/rails/rails/commit/dbda5febe400b0f71c101355df135f21a9a69caf
- https://github.com/rails/rails/commit/a1d236e93110a91d5fd7555cf8f84c8a98680cc3#diff-b8fadd6d1ec091c30f94738b9b193b78ec90681790dd3a6aa79938b4dacbe904

We already used it for the user in the account permissions controller tests in most places, e.g.: https://github.com/alphagov/signon/blob/main/test/controllers/users/permissions_controller_test.rb#L9

My feeling was that (1) it's a bit of syntactic sugar that Rails provides, and Rails is very sugary, and (2) we were already using it, just not consistently, so this makes our use (more?) consistent

Reply:
Thanks for finding those bit of source code, interesting to see where it comes from!

And yeah, I'm not too fussed about this - I do prefer being able to read exactly what's passed in as passing an `application` to `application_id` confuses my previously Type-safe-loving brain a bit but I'm sure I can get over it if it's a common pattern 😅